example-programs/mmap-syscall.c

```diff
@@ -0,0 +1,21 @@
+#include <sys/mman.h>
+#include <sys/stat.h>
+#include <stdio.h>
+#include <fcntl.h>
+#include <unistd.h>
+
+int main() {
+    char *addr;
+    int fd;
+    struct stat sb;
+
+    fd = open("example-programs/mmap-syscall.c", O_RDONLY);
```

Comment:
Please also add error handling to each syscall (you can find an example in `access-itself.c`), so that when the program is run manually, it points out failures if they occur.